### PR TITLE
Add ARF support in "oscap xccdf generate fix"

### PR DESCRIPTION
**Pull request by jan-cerny**
This pull request will enable:

1. Generating fix scripts from a profile in ARF file (profile-oriented fix)
This contains all the remediation scripts from a given profile.
Example:
```
oscap xccdf generate fix \
--profile xccdf_org.ssgproject.content_profile_common \
--output fix.sh \
arf.xml
```

2. Generating fix scripts based on scan results stored in ARF file (result-oriented fix)
This contains remediation only for failed rules.
Example:
```
oscap xccdf generate fix \
--result-id xccdf_org.open-scap_testresult_xccdf_org.ssgproject.content_profile_common \
--output fix.sh \
arf.xml
```

### CODE REVIEW

**Comment by mpreisler** [comment]
CC @rsprudencio

**Comment by mpreisler** [comment]
I'd sleep much better if we had a regression test for this. Looks like this is easy to test automatically.

File: src/XCCDF/xccdf_session.c [hunk 1]
```diff
@@ -159,6 +162,13 @@ struct xccdf_session *xccdf_session_new(const char *filename)
 	return session;
 }
 
+struct xccdf_session *xccdf_session_new(const char *filename)
+{
+	struct oscap_source *source = oscap_source_new_from_file(filename);
+	struct xccdf_session *session = xccdf_session_new_from_source(source);
+	return session;
```
Comment (mpreisler) [inline src/XCCDF/xccdf_session.c]: +1

File: src/XCCDF_POLICY/xccdf_policy_substitute.c [hunk 2]
```diff
@@ -83,6 +83,7 @@ static int _xccdf_text_substitution_cb(xmlNode **node, void *user_data)
 				// process the <xccdf:sub> element as if @use was set to "title". but
 				// during Document Generation or Assessment, process the <xccdf:sub>
 				// element as if @use was set to "value".
+				oscap_free(sub_use);
```
Comment (mpreisler) [inline src/XCCDF_POLICY/xccdf_policy_substitute.c]: This looks like an unrelated memory leak fix.

Comment (jan-cerny) [inline src/XCCDF_POLICY/xccdf_policy_substitute.c]: This memory leak happened to me with generating fix from ARF with `--profile`.

**Comment by jan-cerny** [comment]
I will provide a test for this feature.

**Comment by rsprudencio** [comment]
Awesome work @jan-cerny :smile: 

You'll push the tests to this PR or to a separate PR?

**Comment by jan-cerny** [comment]
@rsprudencio Thanks. I will push it here.

File: tests/API/XCCDF/unittests/test_fix_arf.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+set -e
+set -o pipefail
+
+profile="xccdf_moc.elpmaxe.www_profile_standard"
+result_id="xccdf_org.open-scap_testresult_xccdf_moc.elpmaxe.www_profile_standard"
+bash_line1="echo this_is_ok"
+bash_line2="echo fix_me_please"
+ansible_template="urn:xccdf:fix:script:ansible"
+ansible_line1="this_is_ok"
+ansible_line2="fix_me_please"
```
Comment (dahaic) [inline tests/API/XCCDF/unittests/test_fix_arf.sh]: Can you use different strings for ansible, so they are not substrings of the bash ones?

File: tests/API/XCCDF/unittests/test_fix_arf.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+$OSCAP xccdf generate fix --result-id $result_id --output $script $results_arf >$stdout 2>$stderr
+[ -f $stdout ]; [ ! -s $stdout ]; rm $stdout
+[ -f $stderr ]; [ ! -s $stderr ]; rm $stderr
+grep -q "$bash_line2" $script
+
+# Generate  an Ansible playbook based on scan results stored in ARF file
+$OSCAP xccdf generate fix --result-id $result_id --template $ansible_template --output $playbook $results_arf >$stdout 2>$stderr
+[ -f $stdout ]; [ ! -s $stdout ]; rm $stdout
+[ -f $stderr ]; [ ! -s $stderr ]; rm $stderr
+grep -q "$ansible_line2" $playbook
```
Comment (dahaic) [inline tests/API/XCCDF/unittests/test_fix_arf.sh]: Can you add negative check as well? To make sure there are not lines for other scenarios [ok/not ok, and bash/ansible] present.

**Comment by mpreisler** [comment]
@openscap-jenkins test this please

**Comment by rsprudencio** [comment]
Looks like a legit fail in jenkins ... @openscap-jenkins test this please

File: tests/API/XCCDF/unittests/test_fix_arf.sh (hunk 3 above)
Comment (jan-cerny) [inline tests/API/XCCDF/unittests/test_fix_arf.sh]: @dahaic Yes, I can, that would be definitely better. What do you think should there be a valid ansible playbook or just an arbitrary string? Personally I would prefer a valid ansible playbook but I'm afraid that then it would not be an atomic value.

File: tests/API/XCCDF/unittests/test_fix_arf.sh (hunk 4 above)
Comment (jan-cerny) [inline tests/API/XCCDF/unittests/test_fix_arf.sh]: Yes

**Comment by jan-cerny** [comment]
@rsprudencio Yes, it's legitimate fail. There is a missing `$srcdir` before the XCCDF file name in the test script. So the test works with `make check`, but not with `make distcheck`. I always forgot this :-( I will fix it soon.

**Comment by jan-cerny** [comment]
@openscap-jenkins test this please

**Comment by ybznek** [comment]
test this please

**Comment by rsprudencio** [comment]
@dahaic any other review about those tests? The code looks good to me, if the tests are good too you can merge this or notify me so this PR can get merged. :smile:

**Comment by rsprudencio** [comment]
Mergint this as @dahaic suggestions were fixed.
